dev/releases/update_website.py

```diff
@@ -138,124 +139,112 @@ def mb_bytes(nrbytes):
-try:
-    subprocess.run(["git", "checkout", "-b", branch], check=True)
-except:
-    error("branch " + branch + " already exists")
 
-notice("on git branch " + branch + " of GapWWW")
-notice("using GAP tag " + release['tag_name'])
-notice("using GAP release: " + gap_version)
-notice("using release date: " + date)
-notice("using temporary directory: " + tmpdir)
+notice("Using GAP release: " + gap_version)
+notice("Using temporary directory: " + tmpdir)
 
 
 ################################################################################
-# Download and extract the release asset named 'gap-4.X.Y.tar.gz'
-# TODO perhaps don't hardcode this name?
-tarball = 'gap-' + gap_version + '.tar.gz'
-tarball_url = [ x for x in assets_unix if x['name'] == tarball ]
+# Download and extract the release asset named "gap-4.X.Y.tar.gz"
+tarball = "gap-" + gap_version + ".tar.gz"
+tarball_url = [ x for x in assets_unix if x.name == tarball ]
 try:
-    tarball_url = tarball_url[0]['browser_download_url']
+    tarball_url = tarball_url[0].browser_download_url
 except:
-    error("cannot find " + tarball + " in release at tag " + relrease['tag_name'])
+    error("cannot find " + tarball + " in release at tag " + release.tag_name)
 
-gaproot = tmpdir + 'gap-' + gap_version + '/'
-pkg_dir = gaproot + 'pkg/'
-gap_exe = gaproot + 'bin/gap.sh'
+gaproot = tmpdir + "gap-" + gap_version + "/"
+pkg_dir = gaproot + "pkg/"
+gap_exe = gaproot + "bin/gap.sh"
 
-# TODO error handling
 with working_directory(tmpdir):
-    # TODO only extract if I need to?
-    notice('downloading ' + tarball_url + ' (if not already downloaded) to ' + tmpdir + ' . . .')
+    # TODO Can we sometimes avoid extracting this? Perhaps this script has
+    # already been partially-run, and so we'd be able to skip this step?
+    # TODO error handling
+    notice("Downloading " + tarball_url + " (if not already downloaded) to " + tmpdir + " . . .")
     download_with_sha256(tarball_url, tarball)
-    notice('extracting ' + tarball + ' to ' + gaproot + ' . . .')
+    notice("Extracting " + tarball + " to " + gaproot + " . . .")
     with tarfile.open(tarball) as tar:
-        tar.extractall()
+        try:
+            tar.extractall()
+        except:
+            error("failed to extract tarball (was it already extracted?")
+
+
+################################################################################
+# Extract release date from configure.ac
+if not args.date and not args.use_github_date:
+    with working_directory(gaproot):
+        with open("configure.ac", "r") as configure_ac: 
+            filedata = configure_ac.read()
+            try:
+                # Expect date in YYYY-MM-DD format
+                release_date = re.search('\[gap_releaseday\], \[(\d{4}-\d{2}-\d{2})\]', filedata).group(1)
+            except:
+                error("Cannot find the release date in configure.ac, consider using --date")
+            release_date = datetime.datetime.strptime(release_date, "%Y-%m-%d")
```

Comment:
So before this, `release_date` is a YYYY-MM-DD string, right? Do we really need to parse that? Can't we work with the string below? It's trivial to extract year, month, day from it and thus also the `date` value below. That way, we avoid the dependency on `datetime`. 

It may not be worth the effort to change for now; I am mainly asking to see if there are some deeper difficulties I am missing. (Depending on the answer to that, I will perhaps add a comment here explaining it either way, to stop future me from fretting about it ;-) )

Reply:
Currently, the date can get into this script in three ways:
1. via the argument `--date` (which should be checked is in YYYY-MM-DD format)
2. via configure.ac (which we can rely on being in proper YYYY-MM-DD format)
3. via the option `--use-github-date` (which gets a `datetime` object from the GitHub API)

Currently I convert them all to `datetime` objects to benefit from its validation, and to have them all in a standard format for getting the day, month, year.

For 1, converting to a `datetime` object allows us to avoid having to validate the date ourselves. But of course we could do this ourselves.
For 2, yes we could parse this directly.
For 3, I think we need the `datetime` module to parse this (but I'm not an expert). However, maybe the option`--use-github-date` should be removed anyway.

Reply:
I think validation is a good argument for all of this. Anyway, I am 100% happy to leaves this as-is